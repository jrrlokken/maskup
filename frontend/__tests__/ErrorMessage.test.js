import { configure, shallow } from 'enzyme';
import Adapter from '@hteker/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';
import DisplayError from '../components/ErrorMessage';

configure({ adapter: new Adapter() });

const error = {
  message: 'Error!'
}

describe('<DisplayError/>', () => {
  it('renders properly and matches snapshot', () => {
    const wrapper = shallow(
      <DisplayError error={error} />
    );
    expect(wrapper.find('strong').text()).toBe('Oops!');
    const ptext = wrapper.find('p').text();
    expect(ptext).toBe('Oops!Error!');
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
