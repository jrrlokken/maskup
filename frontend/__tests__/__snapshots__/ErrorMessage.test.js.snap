// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayError/> renders properly and matches snapshot 1`] = `
<ErrorMessage__ErrorStyles>
  <p
    data-test="graphql-error"
  >
    <strong>
      Oops!
    </strong>
    Error!
  </p>
</ErrorMessage__ErrorStyles>
`;
